platform/windows/detect.py

```diff
@@ -725,6 +725,10 @@ def configure_mingw(env: "SConsEnvironment"):
     if os.name == "nt":
         env["TEMPFILEARGESCFUNC"] = tempfile_arg_esc_func
 
+    # FIXME: While MinGW *should* support this limit (documentation is inconsistent), it's only
+    # needed because the GHA for Windows MinGW/GCC fails to recognize the above link method.
+    env["MAXLINELENGTH"] = 32768
```

Comment:
How does this compare with the discussion in #97458 for MSVC? It feels like we're piling workarounds for something that might actually be working properly from SCons if there weren't workarounds? Or is it working correctly only for MSVC and not for MinGW?

I do note that https://www.scons.org/wiki/LongCmdLinesOnWin32 is still in the SCons wiki, but it's pretty old (2015, then just had style updates a couple years ago).

Reply:
MSYS2 shell might support it, but MinGW build can be run from a Windows CMD/PowerShell shell with usual command length limitations, and will cause the same `The command line is too long` issues if we have lines longer than 8K (not sure if MinGW build have any right now).

> it's only needed because the GHA for Windows MinGW/GCC

Is `os.name == "nt"` true in the MSYS2 shell? It might identify as `posix`.

Reply:
`os.name == "nt"` is true on MSYS2 shell, yes. I suspect this is a root of a lot of small annoyances, like the inconsistency of how the RES builder is handled.

I'll rebase a version with the `MAXLINELENGTH` section commented out to showcase the issue.

Reply:
Yep, fails with the following:
```
Using tempfile C:\Users\RUNNER~1\AppData\Local\Temp\tmp0i2g0cx6.rsp for command line:
x86_64-w64-mingw32-gcc-ar rc --thin modules/freetype/libfreetype_builtin.windows.template_release.x86_64.a thirdparty/freetype/src/autofit/autofit.windows.template_release.x86_64.o thirdparty/freetype/src/base/ftbase.windows.template_release.x86_64.o thirdparty/freetype/src/base/ftbbox.windows.template_release.x86_64.o thirdparty/freetype/src/base/ftbdf.windows.template_release.x86_64.o thirdparty/freetype/src/base/ftbitmap.windows.template_release.x86_64.o thirdparty/freetype/src/base/ftcid.windows.template_release.x86_64.o thirdparty/freetype/src/base/ftdebug.windows.template_release.x86_64.o thirdparty/freetype/src/base/ftfstype.windows.template_release.x86_64.o thirdparty/freetype/src/base/ftgasp.windows.template_release.x86_64.o thirdparty/freetype/src/base/ftglyph.windows.template_release.x86_64.o thirdparty/freetype/src/base/ftgxval.windows.template_release.x86_64.o thirdparty/freetype/src/base/ftinit.windows.template_release.x86_64.o thirdparty/freetype/src/base/ftmm.windows.template_release.x86_64.o 
x86_64-w64-mingw32-gcc-ar @C:\Users\RUNNER~1\AppData\Local\Temp\tmp0i2g0cx6.rsp
ERROR: C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ar.exe: unknown option -- @
Usage: C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ar.exe [emulation options] [-]{dmpqrstx}[abcDfilMNoOPsSTuvV] [--plugin <name>] [member-name] [count] archive-file file...
       C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ar.exe -M [<mri-script]
 commands:
  d            - delete file(s) from the archive
  m[ab]        - move file(s) in the archive
  p            - print file(s) found in the archive
  q[f]         - quick append file(s) to the archive
  r[ab][f][u]  - replace existing or insert new file(s) into the archive
  s            - act as ranlib
  t[O][v]      - display contents of the archive
  x[o]         - extract file(s) from the archive
 command specific modifiers:
  [a]          - put file(s) after [member-name]
  [b]          - put file(s) before [member-name] (same as [i])
  [D]          - use zero for timestamps and uids/gids
  [U]          - use actual timestamps and uids/gids (default)
  [N]          - use instance [count] of name
  [f]          - truncate inserted file names
  [P]          - use full path names when matching
  [o]          - preserve original dates
  [O]          - display offsets of files in the archive
  [u]          - only replace files that are newer than current archive contents
 generic modifiers:
  [c]          - do not warn if the library had to be created
  [s]          - create an archive index (cf. ranlib)
  [l <text> ]  - specify the dependencies of this library
  [S]          - do not build a symbol table
  [T]          - deprecated, use --thin instead
  [v]          - be verbose
  [V]          - display the version number
  @<file>      - read options from <file>
  --target=BFDNAME - specify the target object format as BFDNAME
  --output=DIRNAME - specify the output directory for extraction operations
  --record-libdeps=<text> - specify the dependencies of this library
  --thin       - make a thin archive
 optional:
  --plugin <p> - load the specified plugin
 emulation options: 
  No emulation specific options
C:/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ar.exe: supported targets: pe-x86-64 pei-x86-64 pe-bigobj-x86-64 elf64-x86-64 pe-i[386](https://github.com/godotengine/godot/actions/runs/11073846166/job/30771336559?pr=97446#step:10:388) pei-i386 elf32-i386 elf32-iamcu elf64-little elf64-big elf32-little elf32-big srec symbolsrec verilog tekhex binary ihex plugin
 
scons: *** [modules\freetype\libfreetype_builtin.windows.template_release.x86_64.a] Error 1
```

Reply:
Seeing some similar issues crop up in #96407; Gonna revert that commit and see if GCC builds without the need for the `MAXLINELENGTH` hack.

Reply:
Applied a fix suggested in that same PR about commenting out the `gcc-ar` assignment; regular `ar` was enough to get the desired results.

Reply:
> Applied a fix suggested in that same PR about commenting out the `gcc-ar` assignment;

Doesn't that risk breaking cross-compilation from other platforms though?

Reply:
Potentially, but there's no test-cases for cross-compilation atm. Those should probably be setup too, in a followup PR maybe.

I'll see if adding a `os.name == "nt"` check does the trick.